app/models/restaurant_wheel.rb

```diff
@@ -30,17 +40,22 @@ def filtered_restaurants
     @filtered_restaurants
   end
 
-  def ordered_restaurants
-    @ordered_restaurants ||= filtered_restaurants.sort_by(&:average_rating).reverse
+  def total_average_rating
+    @total_average_rating ||= filtered_restaurants.sum(&:average_rating).to_d
   end
 
-  def generate_wheel
-    raise 'Must have restuarants to create a wheel!' if ordered_restaurants.empty?
-
-    @wheel = []
-    while @wheel.count < num_stops
-      @wheel += ordered_restaurants
+  def cummulative_weighted_restaurants
+    total = 0
+    @cummulative_weighted_restaurants ||= filtered_restaurants.map do |restaurant|
+      total += restaurant.average_rating / total_average_rating
```

Comment:
Correct this is weighting the restaurants against each other. It is NOT weighting each restaurants rating based on the number.

If we wanted to do that we should add that to the 'Restaurant#average_rating' method